Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -72,58 +77,56 @@ class Navigator {
-                                      threadsCount: nil,
-                                      endpointConfig: endpointConfig)
-        
-        let historyAutorecordingConfig = [
-            "features": [
-                "historyAutorecording": true
-            ]
-        ]
+        let factory = NativeHandlersFactory(tileStorePath: Self.tilesURL?.path ?? "",
+                                            credentials: Self.credentials ?? Directions.shared.credentials,
+                                            tilesVersion: Self.tilesVersion,
+                                            historyDirectoryURL: Self.historyDirectoryURL)
+        tileVersionState = .nominal
+        tileStore = factory.tileStore
+        historyRecorder = factory.historyRecorder
+        cacheHandle = factory.cacheHandle
+        roadGraph = factory.roadGraph
+        navigator = factory.navigator
         
-        var customConfig = ""
-        if let jsonDataConfig = try? JSONSerialization.data(withJSONObject: historyAutorecordingConfig, options: []),
-           let encodedConfig = String(data: jsonDataConfig, encoding: .utf8) {
-            customConfig = encodedConfig
-        }
-        
-        let configFactory = ConfigFactory.build(for: settingsProfile,
-                                                config: NavigatorConfig(),
-                                                customConfig: customConfig)
-        
-        historyRecorder = HistoryRecorderHandle.build(forHistoryFile: Navigator.historyDirectoryURL?.path ?? "", config: configFactory)
-        
-        let runloopExecutor = RunLoopExecutorFactory.build()
-        cacheHandle = CacheFactory.build(for: tilesConfig,
-                                         config: configFactory,
-                                         runLoop: runloopExecutor,
-                                         historyRecorder: historyRecorder)
+        subscribeNavigator()
+    }
+    
+    /**
+     Destroys and creates new instance of Navigator together with other related entities.
+     
+     Typically, this method is used to restart a Navigator with a specific Version during switching to offline or online modes.
+     - parameter version: String representing a tile version name. `nil` value means "latest". Specifying exact version also enables `fallback` mode which will passively monitor newer version available and will notify `tileVersionState` if found.
+     */
+    func restartNavigator(forcing version: String? = nil) {
+        unsubscribeNavigator()
+        tileVersionState = .nominal
+        navigator.shutdown()
         
-        roadGraph = RoadGraph(MapboxNavigationNative.GraphAccessor(cache: cacheHandle))
+        let factory = NativeHandlersFactory(tileStorePath: Self.tilesURL?.path ?? "",
+                                            credentials: Self.credentials ?? Directions.shared.credentials,
+                                            tilesVersion: Self.tilesVersion,
```

Comment:
Shouldn't it be like this:
```suggestion
                                            tilesVersion: version ?? Self.tilesVersion,
```